addons/stock/models/stock_production_lot.py

```diff
@@ -37,27 +37,47 @@ class ProductionLot(models.Model):
     @api.model
     def generate_lot_names(self, first_lot, count):
         """Generate `lot_names` from a string."""
-        # We look if the first lot contains at least one digit.
-        caught_initial_number = regex_findall(r"\d+", first_lot)
-        if not caught_initial_number:
-            raise UserError(_('The lot name must contain at least one digit.'))
-        # We base the serie on the last number found in the base lot.
-        initial_number = caught_initial_number[-1]
-        padding = len(initial_number)
-        # We split the lot name to get the prefix and suffix.
-        splitted = regex_split(initial_number, first_lot)
-        # initial_number could appear several times, e.g. BAV023B00001S00001
-        prefix = initial_number.join(splitted[:-1])
-        suffix = splitted[-1]
-        initial_number = int(initial_number)
+
+        def int_to_string_base(n, base_alphabet):
+            n_str = ""
+            while True:
+                n, ind = divmod(n, len(base_alphabet))
+                n_str = base_alphabet[ind] + n_str
```

Comment:
As it will start by `ind` `1` in case you reached the end of the alphabet, it's OK for digits as it will start by 1 (so, SN will be `9` ->  `10`), but it doesn't fit well for the letters as it will begin with `b` (so, SN will be `z` -> `ba` instead of `aa`).